src/Microsoft.SqlTools.ServiceLayer/FileBrowser/FileBrowserService.cs

```diff
@@ -0,0 +1,272 @@
+        {
+            this.validatePathsCallbackMap.Add(service, callback);
+        }
+
+        /// <summary>
+        /// Initializes the service instance
+        /// </summary>
+        /// <param name="serviceHost"></param>
+        /// <param name="context"></param>
+        public void InitializeService(ServiceHost serviceHost)
+        {
+            this.ServiceHost = serviceHost;
+
+            // Open a file browser
+            serviceHost.SetRequestHandler(FileBrowserOpenRequest.Type, HandleFileBrowserOpenRequest);
+
+            // Expand a folder node
+            serviceHost.SetRequestHandler(FileBrowserExpandRequest.Type, HandleFileBrowserExpandRequest);
+
+            // Validate the selected files
+            serviceHost.SetRequestHandler(FileBrowserValidateRequest.Type, HandleFileBrowserValidateRequest);
+
+            // Close the file browser
+            serviceHost.SetRequestHandler(FileBrowserCloseRequest.Type, HandleFileBrowserCloseRequest);
+        }
+
+        #region request handlers
+
+        internal async Task HandleFileBrowserOpenRequest(
+            FileBrowserOpenParams fileBrowserParams,
+            RequestContext<bool> requestContext)
+        {
+            try
+            {
+                Task.Run(() => RunFileBrowserOpenTask(fileBrowserParams));
+                await requestContext.SendResult(true);
+            }
+            catch
+            {
+                await requestContext.SendResult(false);
+            }
+        }
+
+        internal async Task HandleFileBrowserExpandRequest(
+            FileBrowserExpandParams fileBrowserParams,
+            RequestContext<bool> requestContext)
+        {
+            try
+            {
+                Task.Run(() => RunFileBrowserExpandTask(fileBrowserParams));
+                await requestContext.SendResult(true);
+            }
+            catch
+            {
+                await requestContext.SendResult(false);
+            }
+        }
+
+        internal async Task HandleFileBrowserValidateRequest(
+            FileBrowserValidateParams fileBrowserParams,
+            RequestContext<bool> requestContext)
+        {
+            try
+            {
+                Task.Run(() => RunFileBrowserValidateTask(fileBrowserParams));
+                await requestContext.SendResult(true);
+            }
+            catch
+            {
+                await requestContext.SendResult(false);
+            }
+        }
+
+        internal async Task HandleFileBrowserCloseRequest(
+            FileBrowserCloseParams fileBrowserParams,
+            RequestContext<FileBrowserCloseResponse> requestContext)
+        {
+            FileBrowserCloseResponse response = new FileBrowserCloseResponse();
+            if (this.ownerToFileBrowserMap.ContainsKey(fileBrowserParams.OwnerUri))
+            {
+                this.ownerToFileBrowserMap.Remove(fileBrowserParams.OwnerUri);
+                response.Succeeded = true;
+            }
```

Comment:
>} [](start = 12, length = 1)

do we need an else clause to set Succeeded to false similar to the other handlers, or is it ok to just use the default Succeeded in this case (anyways, the default would be false for bool, but doesn't hurt to be explicit if that is the intent).

Reply:
I added else clause to make it explicit as I did for other handlers.